### PR TITLE
PIM-5363: Reset the UniqueValueSet after a saveAll 

### PR DESCRIPTION
**Pull request by aRn0D**
There is an issue on imports when you have 2 steps managing the same set of products.
The UniqueValueSet class (stateful) keeps in memory the set of product identifiers, to prevent to not have uniqueness validation errors in a batch of products. This class is not reseted after each step, and throws errors when you try to update the same product in 2 differents steps.

This PR reset the UniqueValueSet after a "save_all" event, to remove the identifiers list after a mass products save.

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | yes
| Added Behats                      | no
| Changelog updated                 | no
| Review and 2 GTM                  | no

### CODE REVIEW

**Comment by juliensnz** [comment]
GTM

**Comment by skeleton** [comment]
squash your commits please.
gtm

File: src/Pim/Bundle/CatalogBundle/EventSubscriber/SaveAllCleanerListener.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ * The UniqueValueSet class is stateful, and used when you import several product, to check if in the product batch
+ * there is no unique identifier issues.
+ * This listener listen the StorageEvents::POST_SAVE_ALL to reset the UniqueValueSet information, to be able to
+ * work in another product batch without uniqueness issues.
+ *
+ * @author    Pierre Allard <pierre.allard@akeneo.com>
+ * @copyright 2016 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class SaveAllCleanerListener
```
Comment (nidup) [inline src/Pim/Bundle/CatalogBundle/EventSubscriber/SaveAllCleanerListener.php]: I would choose a more explicit naming, this one is too "general" imho, maybe, ResetUniqueValidationListener or something like this

File: src/Pim/Bundle/CatalogBundle/Resources/config/event_subscribers.yml [hunk 2]
```diff
@@ -28,59 +29,66 @@ services:
...
+            - '@pim_catalog.repository.product_template'
         tags:
             - { name: kernel.event_subscriber }
+
+    pim_catalog.event_listener.save_all_cleaner:
+        class: '%pim_catalog.event_listener.save_all_cleaner.class%'
+        arguments:
+            - '@pim_catalog.validator.unique_value_set'
+        tags:
+           - { name: kernel.event_listener, event: akeneo.storage.post_save_all }
```
Comment (nidup) [inline src/Pim/Bundle/CatalogBundle/Resources/config/event_subscribers.yml]: why a listener and not a subscriber ?

**Comment by nidup** [comment]
Please change the naming and answer to questions then GTM

**Pull request merged**